jobs/build/aws-ami/Jenkinsfile

```diff
@@ -42,34 +42,38 @@ if ( MOCK.toBoolean() ) {
 }
 
 node(TARGET_NODE) {
-
-    set_workspace()
-
-    stage('build') {
-        try {
-
-            sh(script: '''if [ -d "${WORKSPACE}/openshift-ansible" ]; then 
-                cd ${WORKSPACE}/openshift-ansible ; 
-                git pull; 
-            else
-                git clone https://github.com/openshift/openshift-ansible
-            fi
-
-            cd ${WORKSPACE}
-
-            ''')
-
+    try {
+        set_workspace()
+        def buildlib = null
+        stage('clone') {
+            checkout scm
+            buildlib = load('pipeline-scripts/buildlib.groovy')
+            dir('openshift-ansible') {
+                git 'https://github.com/openshift/openshift-ansible.git'
+            }
+        }
+        stage('venv') {
+            sh '''
+[ -e env/ ] || virtualenv env/
+env/bin/pip install --upgrade pip
+env/bin/pip install --upgrade ansible boto
+'''
+        }
+        stage('build') {
             // create the provisioning_vars.yml file to use as inventory
-            withEnv(["AWS_PROFILE=${WORKSPACE}/.aws/credentials", "PATH=${pwd()}/${PATH}",
-                     "RELEASE_VERSION=${RELEASE_VERSION}", "DEPLOYMENT_TYPE=${DEPLOYMENT_TYPE}",
-                     "BASE_AMI_ID=${BASE_AMI_ID}", "CLUSTERID=${CLUSTERID}",
-                     "AWS_SSH_KEY_USER=${AWS_SSH_KEY_USER}","YUM_BASE_URL=${YUM_BASE_URL}","BASE_AMI_NAME=${BASE_AMI_NAME}"]) {
-                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: 'cloud-provider-creds']]) {
-                    writeFile file:"${WORKSPACE}/provisioning_vars.yml", text:"""---
+            withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: 'cloud-provider-creds']]) {
+                env.CLUSTERID = 'devenv'
+                env.BASE_AMI_ID = 'ami-10aed306'
+                env.AWS_SSH_KEY_USER = 'libra'
```

Comment:
This should be passed in as well I think.  These should be the same as the user that is inside of AWS